### PR TITLE
♻️ convert ha-attributes to lit-element

### PR DESCRIPTION
**Pull request by iantrich**

### CODE REVIEW

File: src/components/ha-attributes.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    const items = this.computeDisplayAttributes(
+      Object.keys(hassAttributeUtil.LOGIC_STATE_ATTRIBUTES).concat(
+        this.extraFilters ? this.extraFilters.split(",") : []
+      )
+    );
+
+    return html`
+      <div class="layout vertical">
+        ${repeat(
```
Comment (bramkragten) [inline src/components/ha-attributes.ts]: What is the reason you use repeat? That is mostly optimal for when you want to change the sorting or add/remove elements dynamic.

File: src/components/ha-attributes.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+class HaAttributes extends LitElement {
+  @property() public stateObj?: HassEntity;
+  @property() public extraFilters?: string;
+
+  protected render(): TemplateResult | void {
+    if (!this.stateObj) {
+      return html``;
+    }
+
+    const items = this.computeDisplayAttributes(
```
Comment (bramkragten) [inline src/components/ha-attributes.ts]: We can just bring this function in line?

File: src/components/ha-attributes.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      )
+    );
+
+    return html`
+      <div class="layout vertical">
+        ${repeat(
+          items,
+          (attribute) => attribute,
+          (attribute) => html`
+            <div class="data-entry layout justified horizontal">
```
Comment (bramkragten) [inline src/components/ha-attributes.ts]: classes `layout justified horizontal` are not defined, are they not needed or missing?

File: src/components/ha-attributes.ts (hunk 1 above)
Comment (iantrich) [reply src/components/ha-attributes.ts]: yeah, should use map...

File: src/components/ha-attributes.ts (hunk 2 above)
Comment (iantrich) [reply src/components/ha-attributes.ts]: sure

File: src/components/ha-attributes.ts (hunk 3 above)
Comment (iantrich) [reply src/components/ha-attributes.ts]: I think they are from the old included styles? maybe? not sure. Will remove.

File: src/components/ha-attributes.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            </div>
+          `
+        )}
+        ${this.stateObj.attributes.attribution
+          ? html``
+          : html`
+              <div class="attribution">
+                ${this.stateObj.attributes.attribution}
+              </div>
+            `}
```
Comment (bramkragten) [inline src/components/ha-attributes.ts]: Turned around

Comment (iantrich) [reply src/components/ha-attributes.ts]: 🙈

Comment (bramkragten) [reply src/components/ha-attributes.ts]: And it is better to return an empty string instead of a html literal.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**